packages/core/types/index.js

```diff
@@ -311,7 +316,9 @@ export interface BaseAsset {
   +isSplittable: ?boolean;
   +isSource: boolean;
   +type: string;
-  +symbols: Map<Symbol, Symbol>;
+  // (symbol exported by this -> name of binding to export)
+  +symbols: $ReadOnlyMap<Symbol, Symbol>;
+  +symbolsLocs: $ReadOnlyMap<Symbol, SourceLocation>;
```

Comment:
What do you think of keeping a single map for this rather than two maps? E.g. Map<Symbol, {identifier: Symbol, loc: SourceLocation}>

Also, should we make these internal and only expose the accessor methods? Might be nice to be able to change the data structures later if needed.

Reply:
> What do you think of keeping a single map for this rather than two maps? E.g. Map<Symbol, {identifier: Symbol, loc: SourceLocation}>

For the internal assets, yes.

> Also, should we make these internal and only expose the accessor methods? Might be nice to be able to change the data structures later if needed.

What about a new `Symbols` type with `setSymbol, getSymbol, hasExportSymbol, hasLocalSymbol, getSymbols`? Otherwise the methods are duplicated between Dependency and Asset.

This would also go well with the fix for https://github.com/parcel-bundler/parcel/issues/4364, where we will need to have a distinction between "asset has no symbol" and "symbols not statically analyzable" (currently this is done via `asset.meta.isCommonJS`, but that cannot be used in `BundleGraph#resolveSymbol`), so `symbols` could be null in that case.

Reply:
Sounds reasonable to me!